Improve wazuh agent documentation (#736)
wazuh/wazuh-agent: docs/dev/run-agent.md

- **Linux:** `/var/lib/wazuh-agent`
- **macOS:** `/Library/Application Support/Wazuh agent.app/var`
- **Windows:** `C:\\ProgramData\\wazuh-agent\\var`

If these directories do not exist at the time of running the enrollment command **when running from sources**, the agent will fail with an error like:

```
Error: Cannot open database: C:\ProgramData\wazuh-agent\var/agent_info.db
```

To avoid this, **manually create the directory** before running the command, based on your operating system:

- **Linux:**
  ```bash
  sudo mkdir -p /var/lib/wazuh-agent
  ```

- **macOS:**
  ```bash
  sudo mkdir -p "/Library/Application Support/Wazuh agent.app/var"
  ```

- **Windows:**
  ```powershell
  mkdir "C:\ProgramData\wazuh-agent\var"
  ```

> Note: If the agent is installed using the installation process, the necessary directories are automatically created.

**Alternative: Use a Custom Data Path**

You can configure a custom location for the agent data by setting the `path.data` option in the `wazuh-agent.yml` configuration file:

```yaml
agent:
  ...
  path.data: "C:/MyCustomPath/"
```

> > Note: If you set `path.data: "."`, the agent will use the current working directory as the data directory.

Reviewer: ```suggestion
> Note: Setting `path.data`: "." will make the agent use the current working directory as the data directory.
```

Default configuration file locations:

- **Linux:** `/etc/wazuh-agent/wazuh-agent.yml`
- **macOS:** `/Library/Application Support/Wazuh agent.app/etc/wazuh-agent.yml`
- **Windows:** `C:\\ProgramData\\wazuh-agent\\config\\wazuh-agent.yml`

### Use a Custom Configuration File

> You can also specify a custom config file when enrolling using the `--config-file` option:

Reviewer: ```suggestion
A custom config file can also be specified during enrollment using the `--config-file` option:
```

```bash
./wazuh-agent --enroll --enroll-url https://localhost:55000 --user <username> --password <password> [--name <agent-name>] --config-file /path/to/wazuh-agent.yml
```

> This is useful if you're testing with different configurations or running the agent from a non-default directory.

Reviewer: ```suggestion
This is useful when testing with different configurations or running the agent from a non-default directory.
```
